tests/base.py

```diff
@@ -0,0 +1,62 @@
+from django.conf import settings
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.sites.models import Site
+
+from cms.models import Page
+from cms.test_utils.testcases import CMSTestCase
+from cms.utils.urlutils import admin_reverse
+
+from djangocms_navigation.constants import SELECT2_CONTENT_OBJECT_URL_NAME
+from djangocms_navigation.test_utils.factories import PageContentFactory
+from djangocms_navigation.test_utils.polls.models import Poll, PollContent
+
+
+class BaseViewTestCase(CMSTestCase):
+    """setup data  for view test case"""
+
+    select2_endpoint = admin_reverse(SELECT2_CONTENT_OBJECT_URL_NAME)
+
+    def setUp(self):
```

Comment:
I still feel this entire class should be deleted as it's just a way to tempt people to create a lot more db objects in their tests than they actually need (how many times will we need 2 users, 2 pages, 2 sites and 2 polls all in one test? do you even need all these objects at the same time in any of your tests?). And these sorts of things do tend to have consequences later on (like tests becoming excruciatingly slow and then people finding them too annoying to regularly run them as a result). Currently, there's simply no reason you can't just define what you need directly in the test case you're using this in.
But if, for some reason, you insist on keeping this class then at least change `setUp` to `setUpTestData` which will run once per TestCase rather than once per each test in the TestCase (as `setUp` does). That will help with not slowing down the tests as much.